ansible_collection/hpe/nimble/plugins/module_utils/hpe_nimble.py

```diff
@@ -236,13 +236,13 @@ def get_pe_id(client_obj, pe_name):
         return resp.attrs.get("id")
 
 
-def get_snapshot_id(client_obj, snap_name):
-    if is_null_or_empty(snap_name):
+def get_snapshot_id(client_obj, vol_name, snap_name):
+    if is_null_or_empty(vol_name) or is_null_or_empty(snap_name):
         return None
     else:
-        resp = client_obj.snapshots.get(name=snap_name)
+        resp = client_obj.snapshots.get(vol_name=vol_name, name=snap_name)
         if resp is None:
-            raise Exception(f"Invalid value for snapshot: {snap_name}")
+            raise Exception(f"No snapshot with name '{snap_name}' found for volume: {vol_name}.")
```

Comment:
nit: `volume {vol_name}`   (Colon not required)

Reply:
k